### PR TITLE
Ignore template creation

### PR DESCRIPTION
**Pull request by spolti**
with template creation:
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.jboss.as.test.integration.auditlog.TLSAuditLogToTCPSyslogTestCase
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 154.367 sec - in org.jboss.as.test.integration.auditlog.TLSAuditLogToTCPSyslogTestCase
Running org.jboss.as.test.integration.beanvalidation.BeanValidationTestCase
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 20.325 sec - in org.jboss.as.test.integration.beanvalidation.BeanValidationTestCase
Running org.jboss.as.test.integration.beanvalidation.hibernate.scriptassert.ScriptAssertTestCase
Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.001 sec - in org.jboss.as.test.integration.beanvalidation.hibernate.scriptassert.ScriptAssertTestCase
Running org.jboss.as.test.integration.beanvalidation.hibernate.validator.BootStrapValidationTestCase
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.842 sec - in org.jboss.as.test.integration.beanvalidation.hibernate.validator.BootStrapValidationTestCase

without:
from logs: INFO: Skipping the environment creation for org.jboss.as.test.integration.auditlog.TLSAuditLogToTCPSyslogTestCase

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.jboss.as.test.integration.auditlog.TLSAuditLogToTCPSyslogTestCase
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 109.862 sec - in org.jboss.as.test.integration.auditlog.TLSAuditLogToTCPSyslogTestCase
Running org.jboss.as.test.integration.beanvalidation.BeanValidationTestCase
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.46 sec - in org.jboss.as.test.integration.beanvalidation.BeanValidationTestCase
Running org.jboss.as.test.integration.beanvalidation.hibernate.scriptassert.ScriptAssertTestCase
Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 sec - in org.jboss.as.test.integration.beanvalidation.hibernate.scriptassert.ScriptAssertTestCase
Running org.jboss.as.test.integration.beanvalidation.hibernate.validator.BootStrapValidationTestCase
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.631 sec - in org.jboss.as.test.integration.beanvalidation.hibernate.validator.BootStrapValidationTestCase

### CODE REVIEW

File: utils/src/main/java/org/jboss/arquillian/ce/utils/TemplateUtils.java [hunk 1]
```diff
@@ -61,12 +61,9 @@ public static void addParameterValues(List<ParamValue> values, Map map, boolean
 
     public static String readTemplateUrl(Template template, TemplateAwareConfiguration configuration, StringResolver resolver) {
         String templateUrl = template == null ? null : template.url();
-        if (templateUrl == null || templateUrl.length() == 0) {
-            templateUrl = resolver.resolve(configuration.getTemplateURL());
```
Comment (alesj) [inline utils/src/main/java/org/jboss/arquillian/ce/utils/TemplateUtils.java]: @spolti where did this resolver usage go?

File: cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java [hunk 2]
```diff
@@ -142,18 +144,24 @@ private void registerRoutes(CECubeConfiguration configuration, OpenShiftClient c
...
 
     private void processTemplate(TestClass tc, OpenShiftAdapter client, CECubeConfiguration configuration)
-        throws DeploymentException {
+            throws DeploymentException {
         final StringResolver resolver = Strings.createStringResolver(configuration.getProperties());
         final Template template = ReflectionUtils.findAnnotation(tc.getJavaClass(), Template.class);
         final String templateURL = readTemplateUrl(template, configuration, resolver);
         final List<? extends OpenShiftResource> resources;
+
+        if (templateURL == null || templateURL.length() == 0) {
```
Comment (alesj) [inline cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java]: Imo, it should be like it was -- readTemplateUrl should fail if it cannot find template.

But I guess we should then better handle cleanup ...

Comment (spolti) [inline cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java]: @alesj the only objective of this PR is continue with the tests if no template is found.
If the template is not provided as annotation or via System Properties the tests will proceed normally only with the runinpod and the cleanup will also work fine after the tests is complete.
The integration tests does not use the template, and doing it we save much time running the integration tests.

Comment (alesj) [inline cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java]: @spolti OK, makes sense.

Any way to determine 100% which tests require template and which don't?
(so we definitely fail on the one's that do require it)

Comment (spolti) [inline cube/src/main/java/org/jboss/arquillian/ce/cube/CEEnvironmentProcessor.java]: @alesj I think it relies only upon who is writing it, if ce-arq doesn't find a template a warning message will be printed, so who is writing this test can use this message to know if a template is needed or not.

**Comment by alesj** [comment]
Added similar handling -- closing this one.

**Pull request closed without merging**